cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -165,39 +152,62 @@ private static void processSlaveCluster() throws Exception {
       System.out.println("Could not read Master Checksum from input file.");
     }
     if (masterTimeMap == null || masterTimeMap.size() == 0) {
-      LOG.info("No region information found in the input file");
+      System.out.println("No region information found in the input file");
     }
 
     checkHBaseReplicationComplete(masterTimeMap);
     //checkSnapshotReplicationComplete(masterChecksum);
   }
 
   private static void checkHBaseReplicationComplete(Map<String, Long> masterTimeMap) throws Exception {
+    boolean complete = true;
+
     // Get replicate Time from table for all regions
     Map<String, Long> slaveTimeMap = getMapFromTable(ReplicationConstants.ReplicationStatusTool.REPLICATE_TIME);
 
-    //Check all regions are accounted on both clusters
+    //Check if all regions are accounted on both clusters
     if (masterTimeMap.size() != slaveTimeMap.size()) {
-      System.out.println("Number of regions on the Master and Slave Clusters do not match."
-                         + " HBase Replication is NOT complete.");
-      return;
+      System.out.println("Number of regions on the Master and Slave Clusters do not match.");
+      complete = false;
     }
 
+    complete = !checkDifferences(masterTimeMap.keySet(), slaveTimeMap.keySet());
+
     //Check the maps for all regions
-    for (Map.Entry<String, Long> writeTimeEntry: masterTimeMap.entrySet()) {
+    for (Map.Entry<String, Long> writeTimeEntry : masterTimeMap.entrySet()) {
       String region = writeTimeEntry.getKey();
-      long writeTime = writeTimeEntry.getValue();
-      long replicateTime = slaveTimeMap.get(region);
+      Long writeTime = writeTimeEntry.getValue();
+      Long replicateTime = slaveTimeMap.get(region);
 
-      if (!isReplicationComplete(masterShutdownTime, writeTime, replicateTime)) {
-        System.out.println("HBase Replication is NOT complete. Region:"
-                             + region + " behind by "
-                             + (writeTime - replicateTime) + "ms.");
-        return;
+      if (replicateTime != null && !isReplicationComplete(masterShutdownTime, writeTime, replicateTime)) {
```

Comment:
github isn't letting me comment on isReplicationComplete since it didn't change in this commit, but I think we should update the check to ` || replicateTime >= writeTime`. The reason is that the write state copied over to the slave may be out of date, and by the time the tool is run, the replicate time might have gone past it.